src/rmp/src/Restructure.cpp

```diff
@@ -160,81 +160,131 @@ void Restructure::runABC()
   files_to_remove.emplace_back(input_blif_file_name_);
 
   // abc optimization
-  bool area_mode = opt_mode_ <= Mode::AREA_3;
+
+  int max_threads = ord::OpenRoad::openRoad()->getThreadCount();
+
+  std::vector<Mode> modes;
+  std::vector<pid_t> child_proc;
+
+  if (opt_mode_ <= Mode::AREA_3) {
+    // Area Mode
+    modes = {Mode::AREA_1, Mode::AREA_2, Mode::AREA_3};
+  } else {
+    // Delay Mode
+    modes = {Mode::DELAY_1, Mode::DELAY_2, Mode::DELAY_3, Mode::DELAY_4};
+  }
+
+  child_proc.resize(modes.size(), 0);
+
   std::string best_blif;
   int best_inst_count = std::numeric_limits<int>::max();
-  int opt_mode_init = static_cast<int>(area_mode)
-                          ? static_cast<int>(Mode::AREA_1)
-                          : static_cast<int>(Mode::DELAY_1);
-  int opt_mode_bound = static_cast<int>(area_mode)
-                           ? static_cast<int>(Mode::AREA_3)
-                           : static_cast<int>(Mode::DELAY_4);
-
-  std::vector<pid_t> child_proc(opt_mode_bound - opt_mode_init + 1, 0);
-
-  for (int opt_mode = opt_mode_init; opt_mode <= opt_mode_bound; opt_mode++) {
-    output_blif_file_name_ = work_dir_name_
-                             + std::string(block_->getConstName())
-                             + std::to_string(opt_mode) + "_crit_path_out.blif";
-
-    opt_mode_ = (Mode) opt_mode;
-    std::string abc_script_file
-        = work_dir_name_ + std::to_string(opt_mode) + "ord_abc_script.tcl";
-    debugPrint(logger_,
-               RMP,
-               "remap",
-               1,
-               "Writing ABC script file {}",
-               abc_script_file);
-    if (writeAbcScript(abc_script_file)) {
-      std::string abc_command = std::string("yosys-abc < ") + abc_script_file;
-      if (logfile_ != "")
-        abc_command = abc_command + " > " + logfile_ + std::to_string(opt_mode);
-
-      pid_t child_pid = fork();
-      if (child_pid == 0) {
-        int ret = std::system(abc_command.c_str());
-        exit(ret);
+
+  debugPrint(logger_,
+             RMP,
+             "remap",
+             1,
+             "Running abc with number of threads = {}",
```

Comment:
@vvbandeira please review messages for the current standard (ie missing period).

Reply:
missing period
ABC should be capitalized since is the tool name

Reply:
suggest Running abc with {} threads.

Reply:
Agree with Cherry, please make the change.